src/components/createOnyxContext.js

```diff
@@ -0,0 +1,47 @@
+import React, {createContext} from 'react';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import Str from 'expensify-common/lib/str';
+import getComponentDisplayName from '../libs/getComponentDisplayName';
+
+const propTypes = {
+    /** Rendered child component */
+    children: PropTypes.node.isRequired,
+};
+
+export default (onyxKeyName) => {
+    const Context = createContext();
+    const Provider = props => (
+        <Context.Provider value={props[onyxKeyName]}>
+            {props.children}
+        </Context.Provider>
+    );
+
+    Provider.propTypes = propTypes;
+    Provider.displayName = `${Str.UCFirst(onyxKeyName)}Provider`;
+
+    const ProviderWithOnyx = withOnyx({
+        [onyxKeyName]: {
+            key: onyxKeyName,
+        },
+    })(Provider);
+
+    const withOnyxKey = (WrappedComponent) => {
+        const Consumer = props => (
+            <Context.Consumer>
+                {(value) => {
+                    const propsToPass = {...props, [onyxKeyName]: value};
+                    return (
+                        // eslint-disable-next-line react/jsx-props-no-spreading
+                        <WrappedComponent {...propsToPass} />
```

Comment:
There are some propType inconvenience due to these dynamic prop names
As you've seen this forces you to write the the ONYX key name as prop (e.g. `reportActionsDrafts_`) in the consumer

Instead it might be handy to leave to consumer name the prop however they like

```jsx
// Optional params to tweak the behavior
const withOnyxKey = ({propName = onyxKeyName, transformValue} = {}) => (WrappedComponent) => {
  /* ... */
  const propsToPass = {...props, [propName]: transformValue ? transformValue(value, props) : value};
  /* ... */
}
```

And then in the wrapper component
```jsx
export default compose(
    withWindowDimensions,
    withReportActionsDrafts({ propName: 'draftMessages' }),
)(ReportActionItem);
```

Or extract the draft message using the `transformValue` func
```jsx
const getDraftMessage = (allDrafts, props) => {
  const {reportID, action} = props;
  const draftKey = `${ONYXKEYS.COLLECTION.REPORT_ACTIONS_DRAFTS}${reportID}_${action.reportActionID}`;        
  const draft = allDrafts[draftKey];
  return draft || '';
}

export default compose(
    withWindowDimensions,
    withReportActionsDrafts({ propName: 'draftMessage', transformValue: getDraftMessage }),
)(ReportActionItem);
```

This should result in no changes to the `draftMessage` prop and its usages

Reply:
Ah nice, that's an interesting idea. Thanks! I was having trouble resolving these propTypes errors.